### PR TITLE
make staging up to date with dev as staging is not updated and behind dev and main

### PR DESCRIPTION
**Pull request by Vamania69**

### CODE REVIEW

File: src/components/Categories.tsx [hunk 1]
```diff
@@ -1,45 +1,51 @@
...
   const [categories, setCategories] = useState([]);
 
   //fetch the categories 
   useEffect(() => {
     const fetchCategories = async () => {
-      const response = await axios.get(`http://localhost:8000/api/categories`);
-      console.log(response);
-      setCategories(response.data.data);
+      try {
+        const response = await axios.get(`${BASE_URL}/categories`);
```
Comment (AnadiSharma007) [inline src/components/Categories.tsx]: move BASE_URL to .env.local file

File: src/components/Hero.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use client'
+
+import Image from 'next/image'
+import React from 'react'
+
+const Hero = () => {
+  return (
+    <section className=''>
+    <div className=''>
+    <Image className='min-h-full min-w-full' src="/Hero.jpg" alt="hero img" height={200} width={100}/>
```
Comment (AnadiSharma007) [inline src/components/Hero.tsx:10]: If you are using static images. Import them as a module and then use it. Like 
`
import heroImage from '/hero.jpg'
<Image className='min-h-full min-w-full' src={heroImage} alt="hero img" height={200} width={100}/>
`

File: src/pages/category/[category].tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+
+  // fetch method
+  useEffect(() => {
+    // async call to fetch products related to the category by category ID
+    const getProductsById = async () => {
+      setIsLoading(true);
+      try {
+        const response = await axios.get(
+          `${BASE_URL}/products?categoryId=${categoryId}`
```
Comment (AnadiSharma007) [inline src/pages/category/[category].tsx]: move to .env.local file

File: src/pages/category/[category].tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <>
+      {!!isLoading ? (
+        <h1 className="text-center ">The product are loading</h1>
+      ) : (
+        <div className="flex flex-wrap p-8 ">
+          {!!categoryProducts &&
+            categoryProducts.map((product) => {
+              return <Card key={product.productId} productDetails={product} />;
```
Comment (AnadiSharma007) [inline src/pages/category/[category].tsx:44]: add ? in product.productId

File: src/utils/getImageUrls.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { iImageProps } from "../interfaces/IProductImageProps";
+export const getImageUrl = ({ productImages }: iImageProps) => {
+  const ImagesUrl = productImages.map((image, i) => {
```
Comment (AnadiSharma007) [inline src/utils/getImageUrls.ts]: add ?.

**Review by AnadiSharma007 — CHANGES_REQUESTED** [review]

File: src/components/Categories.tsx (hunk 1 above)
Comment (Vamania69) [reply src/components/Categories.tsx]: sir i have tried multiple way to handle the error but still the BASE_URL remains undefined also tried after removing the space in the env file

File: src/pages/category/[category].tsx (hunk 3 above)
Comment (Vamania69) [reply src/pages/category/[category].tsx]: similary issuse it consistenty hints on http://localhost:3000/category/undefined/products.....

**Pull request merged**